src/pages/donate/index.tsx

```diff
@@ -1,37 +1,50 @@
+import { CircularProgress, Typography } from "@mui/material";
 import Button from "@mui/material/Button";
 import Card from "@mui/material/Card";
 import CardActions from "@mui/material/CardActions";
 import CardHeader from "@mui/material/CardHeader";
 import Container from "@mui/material/Container";
 import Grid from "@mui/material/Grid";
-import { Streamer } from "@prisma/client";
 import type { NextPage } from "next";
 import Link from "next/link";
-import useSWR from "swr";
+import { useRouter } from "next/router";
+import useStreamers from "~/hooks/useStreamers";
 
 const Donate: NextPage = () => {
-  const { data: streamers = [] } = useSWR<Streamer[]>("/api/streamer");
+  const router = useRouter();
+  const { status, data: streamers, error } = useStreamers();
+
+  if (status === "error" && error instanceof Error) {
+    console.error({ error });
+    router.push("/overview");
```

Comment:
Replace programmatic redirect with a `useEffect`-hook. Side effects shouldn't live inside of the render method.
See: https://reactjs.org/docs/hooks-effect.html